File: ex-5_01/main.cpp
#include "lib/calculator.hpp"


int main(){

    Calculator calculator;
    double a, b;

    std::cout << "Choose an operation: \n";
    std::cout << "1. Addition\n";
    std::cout << "2. Subtraction\n";
    std::cout << "3. Multiplication\n";
    std::cout << "4. Division\n";
    int choice;
    std::cin >> choice;
    std::cout << "" << std::endl;
    std::cout << "Enter the numbers: " << std::endl;
    std::cin >> a;
    std::cin >> b;
    switch (choice)
    {
    case 1:
        std::cout << calculator.add(a, b ) << std::endl;
        break;
    case 2:
        std::cout << calculator.subtract(a, b ) << std::endl;
        break;
    case 3:  
        std::cout << calculator.multiply(a, b ) << std::endl;
        break;
    case 4:
        std::cout << calculator.divide(a, b ) << std::endl;
        break;
    default:
        throw std::invalid_argument("Invalid choice\n");
        break;
    }
    return 0;
}

File: ex-5_01/main.py
import sys
sys.path.append('.')

from exam_lib import Calculator

calculator = Calculator()

choice = input("Choose an operation: \n1. Add\n2. Subtract\n3. Multiply\n4. Divide\n")
a = float(input("Enter the first number: "))
b = float(input("Enter the second number: "))



if choice == '1':
    print(calculator.add(a, b))
elif choice == '2':
    print(calculator.subtract(a, b))
elif choice == '3':
    print(calculator.multiply(a, b))
elif choice == '4':
    print(calculator.divide(a, b))
else:
    print("Invalid choice")

